tests/src/PatternAssertion/BasePatternAssert.php

```diff
@@ -125,6 +127,30 @@ protected function assertElementText($expected, string $selector, Crawler $crawl
     ));
   }
 
+  /**
+   * Asserts text is contained in a particular element.
+   *
+   * @param string|null $expected
+   *   The expected value.
+   * @param string $selector
+   *   The CSS selector to find the element.
+   * @param \Symfony\Component\DomCrawler\Crawler $crawler
+   *   The DomCrawler where to check the element.
+   */
+  protected function assertElementTextContains(?string $expected, string $selector, Crawler $crawler): void {
```

Comment:
I guess this is in line with other assertions, just leaving this as a note. We changed the element attribute assertion to check if the attribute is present or not, so maybe this one too should check if the element has the text or not. For now I think it's fine to leave it like it is.

Reply:
Not sure, this still permits sending an empty string as expected, but my main concern about this is that in our tests cases the actual antagonist of this if a missing element, not an empty value of the attribute

Reply:
True, we can use empty string. Perfect!